### ISSUE TITLE
Additional arguments being ignored

### ISSUE THREAD

**Issue by dai-cb** [garriguv/danger-ruby-swiftformat#12]
My Dangerfile looks like 

```
swiftformat.binary_path = "./Pods/SwiftFormat/CommandLineTool/swiftformat"
swiftformat.check_format
swiftformat.additional_args = "--exclude Pods --disable ident"
```

and yet I'm still seeing in CircleCI / Danger

```
### SwiftFormat found issues:
| File | Rules |
| ---- | ----- |
| Pods/RxTest/RxTest/TestableObserver.swift | indent |
| Pods/RxTest/RxTest/XCTest+Rx.swift | indent |
```

it also seems to be ignoring the `.swiftformat` file in the root which has the same setup.

**Comment by garriguv** [comment]
Try this:

```diff
  swiftformat.binary_path = "./Pods/SwiftFormat/CommandLineTool/swiftformat"
+ swiftformat.additional_args = "--exclude Pods --disable ident"
  swiftformat.check_format
- swiftformat.additional_args = "--exclude Pods --disable ident"
```

Let me know how it goes 😄

**Comment by dai-cb** [comment]
Hmm now failing with `"error running swiftformat: empty output"`

**Comment by dai-cb** [comment]
OK, changed it too:
```
swiftformat.additional_args = "--exclude Pods"
```
And it works - thank you - any insight into why the `.swiftformat` file is being ignored?

**Comment by mgrebenets** [comment]
@dai-cb 
Not sure but maybe an order matters, e.g. `additional_args` need to be set before running `check_format`.

If works for me, I'm passing `swiftformat.additional_args = "--config .swiftformat"` and config if located in the same place as the dangerfile.

The `empty output` happens when using `swiftformat` version `0.37.0` or newer.
Afaik it's fixed on `master` (https://github.com/garriguv/danger-ruby-swiftformat/pull/13) but hasn't been released yet.

**Comment by garriguv** [comment]
> @dai-cb
> Not sure but maybe an order matters, e.g. `additional_args` need to be set before running `check_format`.

That's correct, the order matters and `additional_args` must be set before calling `check_format`.

> If works for me, I'm passing `swiftformat.additional_args = "--config .swiftformat"` and config is located in the same place as the dangerfile.

Maybe `.swiftformat` should be picked up automatically by `swiftformat`? At least setting `--config .swiftformat` works 😄 

I'll close this for now. Thank you for following up, @mgrebenets .

**Comment by dai-cb** [comment]
Still intermittently happening
DangerFile:
```
swiftformat.binary_path = "./Pods/SwiftFormat/CommandLineTool/swiftformat"
swiftformat.additional_args = "--exclude Pods"
swiftformat.check_format(fail_on_error: true)
```
just got this, which should have been excluded

<img width="540" alt="screenshot 2019-01-11 at 14 50 53" src="https://user-images.githubusercontent.com/20546336/51040875-79004900-15b0-11e9-8fc1-654fe18f729a.png">

**Comment by mgrebenets** [comment]
Same for me. The config file I’m passing is ignored and file that should have been ignored are checked.

**Comment by dai-cb** [comment]
@mgrebenets has something changed with the config file, re: https://github.com/garriguv/danger-ruby-swiftformat/issues/12#issuecomment-451358186 - or was the config file's contents always ignored?

**Comment by garriguv** [comment]
Hmmm, this is weird. Does this also happen when you run swiftformat from the command line?

**Comment by dai-cb** [comment]
Yep, excluding seems to work for swiftformat from the command line - you can see 500 odd files are excluded in the dryrun below 
```
$ swiftformat . --dryrun
running swiftformat...
(dryrun mode - no files will be changed)
swiftformat completed. 0/633 files would have been updated in 0.21s

$ swiftformat . --dryrun --exclude Pods
running swiftformat...
(dryrun mode - no files will be changed)
swiftformat completed. 0/113 files would have been updated in 0.08s
```

**Comment by fb55** [comment]
As far as I can tell, this issue stems from this plugin providing a list of files, which means the `--exclude` option will be ignored. An easy workaround would be to allow users to specify their own list of files (defaulting to `find_swift_files`). This is already done by eg. swiftlint.

**Comment by garriguv** [comment]
@fb55 Good catch! I'll have a look at what swiftlint does and come up with something.

**Comment by garriguv** [comment]
In the interest of fast builds, I added an additional `exclude` parameter to the gem. This is an additional layer of filtering on top of only checking added and modified files.

https://github.com/garriguv/danger-ruby-swiftformat/pull/19

This should resolve this issue.

**Comment by garriguv** [comment]
I have released version `0.5.0` of the gem that includes the new `exclude` parameter.
